lib/sensors/gyroscope.cpp

```diff
@@ -0,0 +1,146 @@
+
+      core::Float X_Value = (core::Float)x_axis;
+      X_Value *= 0.00875;
+      const std::optional<core::GyroscopeData> X_values(std::in_place,X_Value,std::chrono::system_clock::now());
+
+        return X_values;
+    }
+    case core::Axis::kY: {
+      const auto gyroscope_y_high_byte = i2c_->readByte(device_address_, kDataYHigh);
+      if (!gyroscope_y_high_byte) {
+        logger_.log(
+          core::LogLevel::kFatal, "Failed to read gyroscope Y-axis high at channel %d", channel_);
+        return std::nullopt;
+      }
+      const auto gyroscope_y_low_byte = i2c_->readByte(device_address_, kDataYLow);
+      if (!gyroscope_y_low_byte) {
+        logger_.log(
+          core::LogLevel::kFatal, "Failed to read gyroscope Y-axis low at channel %d", channel_);
+        return std::nullopt;
+      }
+      const auto y_axis = ((*gyroscope_y_high_byte << 8) | *gyroscope_y_low_byte);
+      logger_.log(
+        core::LogLevel::kDebug, "Successfully read y-axis from gyroscope at channel %d", channel_);
+
+      core::Float Y_Value = (core::Float)y_axis;
+      Y_Value *= 0.00875;
+      const std::optional<core::GyroscopeData> Y_values(std::in_place,Y_Value,std::chrono::system_clock::now());
+
+        return Y_values;
+    }
+    case core::Axis::kZ: {
+      const auto gyroscope_z_high_byte = i2c_->readByte(device_address_, kDataZHigh);
+      if (!gyroscope_z_high_byte) {
+        logger_.log(
+          core::LogLevel::kFatal, "Failed to read gyroscope Z-axis high at channel %d", channel_);
+        return std::nullopt;
+      }
+      const auto gyroscope_z_low_byte = i2c_->readByte(device_address_, kDataZLow);
+      if (!gyroscope_z_low_byte) {
+        logger_.log(
+          core::LogLevel::kFatal, "Failed to read gyroscope Z-axis low at channel %d", channel_);
+        return std::nullopt;
+      }
+      const auto z_axis = ((*gyroscope_z_high_byte << 8) | *gyroscope_z_low_byte);
+      logger_.log(
+        core::LogLevel::kDebug, "Successfully read z-axis from gyroscope at channel %d", channel_);
+
+      core::Float Z_Value = (core::Float)z_axis;
+      Z_Value *= 0.00875;
+      const std::optional<core::GyroscopeData> Z_values(std::in_place,Z_Value,std::chrono::system_clock::now());
+
+        return Z_values;
+    }
+    default: {
+      logger_.log(core::LogLevel::kFatal, "Gave an invalid axis");
+      return std::nullopt;
+    }
+  }
+}
+
+std::optional<Gyroscope> Gyroscope::create(core::ILogger &logger,
+                                           std::shared_ptr<io::II2c> i2c,
+                                           const std::uint8_t channel,
+                                           const std::uint8_t device_address)
+{
+  const core::Result write_result_from_ctrl1
+    = i2c->writeByteToRegister(device_address, kCtrl1, kConfigurationSetting1);
+  if (write_result_from_ctrl1 == core::Result::kFailure) {
+    logger.log(
+      core::LogLevel::kFatal,
```

Comment:
Again, isn't this only critical failure if we're actually using active suspension?